src/Authoring/WinRT.SourceGenerator/DiagnosticHelpers.cs

```diff
@@ -330,6 +353,111 @@ private string SimplifySyntaxTypeString(string syntaxType)
             }
         }
 
+        private SpecialType[] AllowedStructFieldTypes = new SpecialType[] 
+        {
+            SpecialType.System_Boolean, 
+            SpecialType.System_String,
+            SpecialType.System_Single,
+            SpecialType.System_Double,
+            SpecialType.System_UInt16,
+            SpecialType.System_UInt32,
+            SpecialType.System_UInt64,
+            SpecialType.System_Int16,
+            SpecialType.System_Int32,
+            SpecialType.System_Int64,
+            SpecialType.System_Enum,
+        };
+
+        private static readonly Dictionary<string, List<string>> MappedCSharpTypeMethods = new Dictionary<string, List<string>>()
+        {
+            { "System.DateTimeOffset", new List<string> { "Add", "AddDays", "AddHours", "AddMilliseconds", "AddMinutes", "AddMonths", "AddSeconds", "AddTicks", "AddYears", "Compare", "CompareTo",
+                "Equals", "EqualsExact", "FromFileTime", "FromUnixTimeMilliseconds", "FromUnixTimeSeconds", "GetHashCode", "Parse", "ParseExact", "Subtract", "ToFileTime", "ToLocalTime",
+                "ToOffset", "ToString", "ToUniversalTime", "ToUnixTimeMilliseconds", "ToUnixTimeSeconds", "TryFormat", "TryParse", "TryParseExact" }  },
+            { "System.Exception", new List<string> { "GetBaseException", "GetObjectData", "GetType", "ToString" } },
+            { "System.IDisposable", new List<string> { "Dispose" } },

```

Comment:
Any way to make the method lookup generic.  I am concerned about overloaded methods with different # of parameters that are from the component.

Reply:
Addressed